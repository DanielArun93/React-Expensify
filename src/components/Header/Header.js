import React from 'react';
import ReactDOM from 'react-dom';
import { NavLink } from 'react-router-dom';


class Header extends React.Component {
    render() {
        return (
            <div>
                <h1>Expensify</h1>
                <NavLink to="/" activeClassName="is-active" exact>Home</NavLink><br />
                <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink><br />
                <NavLink to="/edit/:id" activeClassName="is-active">Edit Expense</NavLink><br />
                <NavLink to="/help" activeClassName="is-active">Help</NavLink>
            </div>
        )
    }
}

export default Header;