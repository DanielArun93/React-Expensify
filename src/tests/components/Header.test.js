import React from 'react';
import { shallow } from 'enzyme';
//import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header/Header';



// test('should render header Component',() => {
//     const renderer = new ReactShallowRenderer();
//     renderer.render(<Header />);
//     expect(renderer.getRenderOutput()).toMatchSnapshot();
//     //console.log(renderer.getRenderOutput());
// })

test('should render header Component', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text()).toBe("Expensify");
})