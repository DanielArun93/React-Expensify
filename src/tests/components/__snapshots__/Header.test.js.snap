// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header Component 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/create"
  >
    Create Expense
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/edit/:id"
  >
    Edit Expense
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/help"
  >
    Help
  </NavLink>
</div>
`;
